lib/goldiloader/custom_preloads.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Goldiloader
+  module CustomPreloads
+    def initialize
+      super
+      @custom_preloads = nil
+    end
+
+    def preloaded(model, cache_name:, key:, &block)
+      unless preloaded?(cache_name)
+        ids = models.map do |record|
+          record.public_send(key)
+        end
+
+        # We're using instance_exec instead of a simple yield to make sure that the
+        # given block does not have any references to the model instance as this might
+        # lead to unexpected results
+        preloaded_hash = instance_exec(ids, &block)
```

Comment:
Good catch. Executing the block in the context of a single model instance would definitely result in strange behavior but executing it in the context of an `AutoIncludeContext` could also lead to strange results. Would it make sense to execute the block in the context of the model class?

Reply:
Good point! I just looked for a context that is _not_ the instance. Using the model class obviously makes more sense 👍